pkg/virt-launcher/notify-client/client.go

```diff
@@ -419,12 +431,15 @@ func (n *Notifier) StartDomainNotifier(
 					}
 				}
 			case agentUpdate := <-agentStore.AgentUpdated:
-				interfaceStatuses = agentUpdate.DomainInfo.Interfaces
-				guestOsInfo = agentUpdate.DomainInfo.OSInfo
-				fsFreezeStatus = agentUpdate.DomainInfo.FSFreezeStatus
+				guestEvent.interfaceStatuses = agentUpdate.DomainInfo.Interfaces
+				guestEvent.guestOsInfo = agentUpdate.DomainInfo.OSInfo
+				guestEvent.fsFreezeStatus = agentUpdate.DomainInfo.FSFreezeStatus
+
+				eventCallback(domainConn, domainCache, libvirtEvent{}, n, deleteNotificationSent, guestEvent, vmi)
+			case domainUpdate := <-domainUpdateChan:
+				guestEvent.domainMemoryDumpInfo = domainUpdate.MemoryDumpInfo
+				eventCallback(domainConn, domainCache, libvirtEvent{}, n, deleteNotificationSent, guestEvent, vmi)
```

Comment:
i'm unsure about how this works. It doesn't appear like the memory dump info is persisted on the Domain, so if this event doesn't get processed as expected by virt-handler, I think we'd never have a chance to get the result of the memory dump again.

The method we have for storing persistent info on the domainSpec is the `domainSpec.Metadata` field.

Reply:
This is a really good point! I am fixing it to get if the memory dump completed from the memoryDump call by updating the info of the last memory dump on the LibvirtDomainManager struct.